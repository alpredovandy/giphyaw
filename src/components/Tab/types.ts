export interface TabProps {
    onType: (type: string) => void;
    value: string;
    items: { title: string; value: string }[];
}
