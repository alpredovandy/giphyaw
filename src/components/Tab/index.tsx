import { FC } from 'react';

import { TabProps } from '@/components/Tab/types.ts';

const Tab: FC<TabProps> = ({ onType, value, items = [] }) => (
    <div role="tablist" className="tabs tabs-boxed">
        {items.map((item) => (
            <a key={item.value} role="tab" className={`tab ${value === item.value ? 'tab-active' : ''}`} onClick={() => onType(item.value)}>
                {item.title}
            </a>
        ))}
    </div>
);

export default Tab;
