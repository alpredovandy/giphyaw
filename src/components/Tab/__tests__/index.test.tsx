import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';

import Tab from '@/components/Tab';
import type { TabProps } from '@/components/Tab/types';

describe('Test components: Tab', () => {
    let props: TabProps;
    const renderTab = () => render(<Tab {...props} />);

    it('Snapshot', () => {
        const { container } = renderTab();

        expect(container).toMatchSnapshot();
    });
});
