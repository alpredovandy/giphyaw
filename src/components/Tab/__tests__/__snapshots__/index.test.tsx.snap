// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Test components: Tab > Snapshot 1`] = `
<div>
  <div
    class="tabs tabs-boxed"
    role="tablist"
  />
</div>
`;
